project/src/com/google/daggerquery/executor/services/SourcesLoader.java

```diff
@@ -28,12 +29,13 @@
   private static final String PATH_TO_BINDING_GRAPH = "/com/google/daggerquery/binding_graph_data.textproto";
 
   public BindingGraph loadBindingGraph() throws IOException {
-    InputStream inputStream = SourcesLoader.class.getResourceAsStream(PATH_TO_BINDING_GRAPH);
+    try (InputStream inputStream = SourcesLoader.class.getResourceAsStream(PATH_TO_BINDING_GRAPH))  {
+      if (inputStream == null) {
+        throw new FileNotFoundException(String.format("File %s is missing.", PATH_TO_BINDING_GRAPH));
+      }
 
-    if (inputStream == null) {
-      throw new FileNotFoundException(String.format("File %s is missing.", PATH_TO_BINDING_GRAPH));
+      int sizeOfBindingGraph = ByteBuffer.wrap(inputStream.readNBytes(4)).getInt();
```

Comment:
Where does the constant `4` come from?

Reply:
It's the number of bytes required to save an integer (i.e. 32 / 8). 

Actually I'm not sure that 2 147 483 647 (max int value) is enough to store the huge graphs in AGSA and I don't know how to check it now.